packages/completer/src/manager.ts

```diff
@@ -172,6 +180,9 @@ export class CompletionProviderManager implements ICompletionProviderManager {
       // Create a new handler.
       const handler = await this._generateHandler(newCompleterContext, options);
       this._panelHandlers.set(widget.id, handler);
+      handler.completer.selected.connect((sender, value) =>
+        this._selected.emit(value)
```

Comment:
No objections to your approach in principle. The only thing that I would change is that the emit type ideally would not be a string, but an an object following an interface (which could contain the string value) so that we could later add more properties without making a breaking change. Maybe:  `Required<Pick<ICompletionItem, 'insertText'>>`.

Reply:
that's a great idea, I'll update it

Reply:
pushed the update, feel free to check it, I decided to call the interface `ICompleterSelection` since it is a selection coming from a completer. I also added the actual `Completer` object to the interface, but decided that if we ever need that in the future it can easily be added.